mitiq/zne/scaling/tests/test_layer_scaling.py

```diff
@@ -53,24 +60,59 @@ def test_layer_folding():
     # Iterate over every possible combination of layerwise folds for a maximum
     # number of 5-folds.
     total_folds = 5
-    for i1 in range(total_folds):
-        for i2 in range(total_folds):
-            for i3 in range(total_folds):
-                layers_to_fold = [i1, i2, i3]
-
-                folded_circuit = layer_folding(circ, layers_to_fold)
-
-                # For a given layer, the number of copies on a layer will be
-                # 2n + 1 where "n" is the number of folds to perform.
-                qreg = LineQubit.range(3)
-                correct = Circuit(
-                    # Layer-1
-                    [ops.H.on_each(*qreg)] * (2 * (layers_to_fold[0]) + 1),
-                    # Layer-2
-                    [ops.CNOT.on(qreg[0], qreg[1])]
-                    * (2 * (layers_to_fold[1]) + 1),
-                    [ops.X.on(qreg[2])] * (2 * (layers_to_fold[1]) + 1),
-                    # Layer-3
-                    [ops.TOFFOLI.on(*qreg)] * (2 * (layers_to_fold[2]) + 1),
-                )
-                assert folded_circuit == correct
+    for i, j, k in product(range(total_folds), repeat=3):
```

Comment:
nit (and very much about personal taste): I like the use of product, but it's arguable that i,j,k are better indices than numbered ones, given we are talking about an ordered number of layers here

Reply:
Fair point. Changed back in https://github.com/unitaryfund/mitiq/pull/2545/commits/c9fb3bcc936807e111af54e7cac801a10532720e.